src/components/PersonsDropDown/PersonsDropDown.tsx

```diff
@@ -0,0 +1,98 @@
+import React, {
+  useCallback, useMemo, useState,
+} from 'react';
+import { Person } from '../../types/Person';
+import Search from '../Search/Search';
+import { debounce } from '../../helpers/helpers';
+import { peopleFromServer } from '../../data/people';
+import { DEFAULT_DELAY } from '../../constants/default.constants';
+import PersonItem from '../PersonItem/PersonItem';
+
+const getUserByName = (
+  users: Person[],
+  query: string,
+) => {
+  let preparedUsers = [...users];
+
+  if (query) {
+    const queryToLower = query.toLowerCase().trim();
+
+    preparedUsers = preparedUsers.filter(({ name }) => (
+      name.toLowerCase().includes(queryToLower)
+    ));
+  }
+
+  return preparedUsers;
+};
+
+type Props = {
+  setSelectedPerson: (person: Person | null) => void;
+  delay?: number,
+};
+
+const PersonsDropDown: React.FC<Props> = ({
+  setSelectedPerson,
+  delay,
+}) => {
+  const [searchValue, setSearchValue] = useState('');
+  const [hasFocusField, setHasFocusField] = useState(false);
+  const [appliedSearchValue, setAppliedSearchValue] = useState('');
+
+  const preparedPersons = useMemo(() => {
+    return getUserByName(peopleFromServer, appliedSearchValue);
+  }, [peopleFromServer, appliedSearchValue]);
+
+  const applySearchValue = useCallback(
+    debounce(setAppliedSearchValue, delay || DEFAULT_DELAY),
+    [],
+  );
+
+  const handleInputField = useCallback((
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setSearchValue(event.target.value);
+    applySearchValue(event.target.value);
+  }, []);
+
+  const handleUserSelect = (selected: Person) => {
+    const preparedPerson = preparedPersons
+      .find(({ slug }) => slug === selected.slug) ?? null;
+
+    setSelectedPerson(preparedPerson);
+    setSearchValue(preparedPerson?.name || '');
+    setAppliedSearchValue(preparedPerson?.name || '');
+  };
+
+  return (
+    <div className="dropdown is-active">
+      <Search
+        value={searchValue}
+        handleInputField={handleInputField}
+        setHasFocusField={setHasFocusField}
+      />
+
+      {hasFocusField && (
+        <div
+          style={{ width: '100%' }}
+          className="dropdown-menu"
+          role="menu"
+        >
+          <div className="dropdown-content">
+            {Boolean(preparedPersons.length)
+              && preparedPersons.map((person: Person) => (
+                <PersonItem
+                  key={person.slug}
+                  person={person}
+                  handleUserSelect={handleUserSelect}
+                />
+              ))}
+
+            {!preparedPersons.length && <p>No matching results</p>}
```

Comment:
https://github.com/mate-academy/react_autocomplete/pull/247#discussion_r1326116239 - it is still not correct.
An example of work can be found in the link in the task:
https://gyazo.com/7956c0cfcebb983ee5c1f89d0847922f